eslint.config.js

```diff
@@ -129,33 +130,31 @@ module.exports = [
     {
         files: ["lib/rules/*", "tools/internal-rules/*"],
         ignores: ["**/index.js"],
-        rules: {
-            ...eslintPlugin.configs["rules-recommended"].rules,
-            "eslint-plugin/no-missing-message-ids": "error",
-            "eslint-plugin/no-unused-message-ids": "error",
-            "eslint-plugin/prefer-message-ids": "error",
-            "eslint-plugin/prefer-placeholders": "error",
-            "eslint-plugin/prefer-replace-text": "error",
-            "eslint-plugin/report-message-format": ["error", "[^a-z].*\\.$"],
-            "eslint-plugin/require-meta-docs-description": ["error", { pattern: "^(Enforce|Require|Disallow) .+[^. ]$" }],
-            "internal-rules/no-invalid-meta": "error"
-        }
+        ...merge({}, eslintPluginRulesRecommendedConfig, {
+            rules: {
+                "eslint-plugin/prefer-placeholders": "error",
+                "eslint-plugin/prefer-replace-text": "error",
+                "eslint-plugin/report-message-format": ["error", "[^a-z].*\\.$"],
+                "eslint-plugin/require-meta-docs-description": ["error", { pattern: "^(Enforce|Require|Disallow) .+[^. ]$" }],
+                "internal-rules/no-invalid-meta": "error"
+            }
+        })
     },
     {
         files: ["lib/rules/*"],
-        ignores: ["index.js"],
+        ignores: ["**/index.js"],
         rules: {
             "eslint-plugin/require-meta-docs-url": ["error", { pattern: "https://eslint.org/docs/latest/rules/{{name}}" }]
         }
     },
     {
         files: ["tests/lib/rules/*", "tests/tools/internal-rules/*"],
-        rules: {
-            ...eslintPlugin.configs["tests-recommended"].rules,
-            "eslint-plugin/prefer-output-null": "error",
```

Comment:
This rule is now already in the tests recommended config, so I removed it from here.